src/librustc_codegen_llvm/debuginfo/type_names.rs

```diff
@@ -98,7 +98,7 @@ pub fn push_debuginfo_type_name<'a, 'tcx>(cx: &CodegenCx<'a, 'tcx>,
         ty::Array(inner_type, len) => {
             output.push('[');
             push_debuginfo_type_name(cx, inner_type, true, output);
-            output.push_str(&format!("; {}", len.unwrap_usize(cx.tcx)));
+            output.push_str(&format!("; {}", len.unwrap_evaluated().unwrap_usize(cx.tcx)));
```

Comment:
Might be worth having `unwrap_usize` on `LazyConst`? You also have `assert_usize`, after all.